### ISSUE TITLE
POS Printer Plugin 

### ISSUE THREAD

**Issue by daymerfillad** [capacitor-community/proposals#36]
Some plugin to print from POS printers from Capacitor

**Comment by IT-MikeS** [comment]
Can you provide more details?
Such as:
- Make/Model of the printer
- Exsisting API for the printer
- Ect.

**Comment by daymerfillad** [comment]
Hello @IT-MikeS my idea is that there is some plugin capable of printing in various types of POS printers. Something similar to https://github.com/mike42/escpos-php or https://github.com/DantSu/ESCPOS-ThermalPrinter-Android/tree/master/escposprinter

**Comment by daymerfillad** [comment]
https://ionicframework.com/docs/native/printer

**Comment by sta55en** [comment]
Just throwing my 2c at this...

I have a lot of experience in implementing printing through Capacitor/Cordova on iOS and Android for Star and Brother printers; and I don't feel like a one-size-fits-all solution is going to work very well here. 

There's a big difference in how various printers communicate across different platforms and building a single plugin will no doubt end up causing a situation where the plugin gets way too bloated and way too heavy trying to do everything for everyone. This is the bane of most Cordova plugins' existence. Do one thing and do it well. So if we were thinking about this, it makes more sense to build a Star Printer plugin, and a Brother one, and a Citizen one, etc.

Some factors to consider when building a plugin like the one proposed:

Which control languages are going to be supported Esc/P, Esc/POS, Star, and some other proprietary formats.  Or are we just connecting to the printer and allowing the library consumer to send through a string?

How will the plugin connect to the printer? BT or BTLE? Very different beasts with very different requirements. 

Which printers will be supported? If we're connecting to the device via a serial connection, we need to understand the device classes and capabilities being supported.

The last issue is that some printers have SDKs that require OEM certificates and manufacturer specific libraries (Both Brother and Star on iOS) - so without bundling those (which might not be allowed by the TOS) it would be a half baked solution that still requires manual installation. 

It feels like building specific plugins for specific devices makes more sense in this case, rather than a blanket plugin.

**Comment by daymerfillad** [comment]
Hello @sta55en thanks for your input

**Comment by rdlabo** [comment]
I created one of POS printer-plugin: https://github.com/rdlabo-team/capacitor-brotherprint

I think @sta55en opinion is right. third-party library is very heavy weight. Many POS printer support plugin will be difficult to use.

**Comment by vitalijalbu** [comment]
> https://ionicframework.com/docs/native/printer

How to integrate this in react? Is it possible?? I need to print receipes to thermal printers... Pls help. Thanks.

**Comment by vitalijalbu** [comment]
> printing

It's possible to integrate a thermal printer in ionic + react + capacitor?? I can't find nothing on internet... If you can help me I can pay as well... :D

**Comment by sta55en** [comment]
Yup, but you would need specific code for each brand of thermal printer.
there are different printing standards, so check out ESCPOS. It is the most
common one.

**Comment by vitalijalbu** [comment]
For all you guys i have a native plugin for pos printer (built using capacitor). If interested DM me.

**Comment by riderx** [comment]
@rdlabo i added your plugin to [awesome-capacitor](https://github.com/riderx/awesome-capacitor) if that is useful please take time to star the repo as well :)

**Comment by WingsDevelopment** [comment]
@vitalijalbu Hey man, I am in same situation here, i need to integrate POS therminal printer in ionic + react + capacitor... Did you find solution? What did you do ?

**Comment by vitalijalbu** [comment]
> @vitalijalbu Hey man, I am in same situation here, i need to integrate POS therminal printer in ionic + react + capacitor... Did you find solution? What did you do ?

hi yes, we bought a license from a german company to archive that, 2 years ago...

**Comment by WingsDevelopment** [comment]
wow, can you share a link to that company ? is that manufacturer of POS therminal you used?

**Comment by vitalijalbu** [comment]
***@***.***/capacitor-cidprint





Il giorno sab 12 ago 2023 alle 19:09 WingsDevelopment <
***@***.***> ha scritto:

**Comment by vitalijalbu** [comment]
I think I also need to have somewhere a jsx file with some buttons where to
activate the plugin and activate Bluetooth and make it run, will share it
give me your email.

Il giorno sab 12 ago 2023 alle 19:09 WingsDevelopment <
***@***.***> ha scritto:

**Comment by WingsDevelopment** [comment]
bans.cowboy@gmail.com thanks alot!
